src/sdam/description/server.rs

```diff
@@ -19,11 +19,11 @@ const DRIVER_MAX_WIRE_VERSION: i32 = 7;
 pub enum ServerType {
     Standalone,
     Mongos,
-    RSPrimary,
-    RSSecondary,
-    RSArbiter,
-    RSOther,
-    RSGhost,
+    RsPrimary,
```

Comment:
I think these are technically breaking changes since this type is exposed through the selection criteria API, but since we're doing a major release, that's not an issue

Reply:
It might be worth filing a ticket then for this change so that it'll be represented in the release notes.

Reply:
Filed [RUST-719](https://jira.mongodb.org/browse/RUST-719), going to split this PR into two commits so the ticket number will show up in our log.